src/api/app/views/webui2/webui/repositories/_repository_entry.html.haml

```diff
@@ -3,12 +3,11 @@
     .card-header
       = link_to(repository, project_repository_state_path(project: project, repository: repository.name), class: 'font-weight-bold')
       .small
-        - if repository.architectures.size.zero?
+        - if repository.architectures.blank?
           No architecture selected
         - else
-          #{repository.architectures.pluck(:name).join(', ')}
+          #{repository.architectures.map(&:name).join(', ')}
```

Comment:
Are you sure? `map` will always trigger the collection proxy* and convert the result to an array, after fetching all attributes. `pluck(:name)` would only fetch the `name` attributes and if I am not mistaken would use cached data from the proxy when available.

[*] similar to `size`

Reply:
We could check once I can preload architectures - but what I'm seing is a select name ... for every repository from within the view. I don't see this anymore once the map is there.

Reply:
@coolo we need a Benchmark which includes the Ruby code here :wink:

Reply:
What we need are preloaded architectures - the rest is to be seen :)